sky/provision/instance_setup.py

```diff
@@ -297,6 +287,64 @@ def start_ray_on_head_node(cluster_name: str, custom_resource: Optional[str],
         'RAY_worker_maximum_startup_concurrency=$(( 3 * $(nproc --all) )) '
         f'{constants.SKY_RAY_CMD} start --head {ray_options} || exit 1;' +
         _RAY_PRLIMIT + _DUMP_RAY_PORTS + RAY_HEAD_WAIT_INITIALIZED_COMMAND)
+    return cmd
+
+
+def ray_worker_start_command(custom_resource: Optional[str],
+                             custom_ray_options: Optional[Dict[str, Any]],
+                             no_restart: bool) -> str:
+    """Returns the command to start Ray on the worker node."""
+    ray_options = (
+        f'--address=${{SKYPILOT_RAY_HEAD_IP}}:{constants.SKY_REMOTE_RAY_PORT} '
```

Comment:
@Michaelvll This need to use the ray port from env var retrieved from head node, to keep the backward compatibility.

Reply:
Fixed